### ISSUE TITLE
Exams, Exercises, Statistics can still be accessed after course end with link

### ISSUE THREAD

**Issue by julian-christl** [ls1intum/Artemis#2075]
### Describe the bug
When the course ends the student should not be able anymore to access it. However if you have the link it's still possible

#### To Reproduce
1. Create a course which ends soon
2. Copy the link to e.g. the exams tab
3. Wait for the course to end
4. Make sure you can't see the course anymore in the dashboard
5. Try to access the link you copied.

#### Expected behavior
You get either a 404 (Not found) or 409 (Conflict) since you are not allowed to access it anymore. Should not apply to instructors but a studend reported this.

**Comment by MaisiKoleni** [comment]
I knew this was the case but couldn't really think of a reason why disabling this could be necessary. Do you have anything special in mind? (e.g. on Moodle, students can always view their old courses, which didn't pose problems as far as I know.)

**Comment by julian-christl** [comment]
I think it's a bad design leaving a link accessible while hiding it in the UI.
Either you make this a feature by not necessarily having to set a end date for the course (if not already possible) then this issue doesn't arise - or you block it.
It shouldn't be such a shady way like now.

**Comment by MaisiKoleni** [comment]
Maybe we should allow the students to see their inactive courses then, similar to the instructor/TA view or Moodle.

**Open question to other instructors: what would your expect or prefer?**

**Comment by julian-christl** [comment]
Yes, maybe introducing a new status "inactive" which will start once the course ends and in which students can see everything but edit/report/etc. anything.
